docs/concepts/submit-data.md

```diff
@@ -0,0 +1,221 @@
+already expressed interest in pursuing or accepting some flavor of a protocol
+enforced minimum fee.
+
+As of version v1.0.0 of the application (celestia-app), gas is not refunded.
+Instead, transaction fees are deducted by a flat fee, originally determined by
+the user via multiplying the gas limit by the desired gas price. This means that
+users should use an accurate gas limit value if they do not wish to over pay.
+Again, community members have already expressed pursuing and accepting a
+solution to this problem.
+
+#### Estimating PFB Gas
+
+Generally, the gas used by a PFB transaction involves a static "fixed cost" and
+a dynamic cost based on the size of each blob involved in the transaction.
+
+> Note: For a general use case of a normal account submitting a PFB, the static
+> costs can be treated as such. However, due to the description above of how gas
+> works in the cosmos-sdk this is not always the case. Notably, if we use a
+> vesting account or the `feegrant` modules, then these static costs change.
+
+The "fixed cost" is an approximation of the gas consumed by operations outside
+the function `GasToConsume` (for example, signature verification, tx size, read
+access to accounts), which has a default value of 65,000.
+
+> Note: the first transaction sent by an account (sequence number == 0) has an
+> additional one time gas cost of 10,000. If this is the case, this should be
+> accounted for.
+
+Each blob in the PFB contributes to the total gas cost based on its size. The
+function `GasToConsume` calculates the total gas consumed by all the blobs
+involved in a PFB, where each blob's gas cost is computed by first determining
+how many shares are needed to store the blob size. Then, it computes the product
+of the number of shares, the number of bytes per share, and the `gasPerByte`
+parameter. Finally, it adds a static amount per blob.
+
+The gas cost per blob byte and gas cost per transaction byte are parameters that
+could potentially be adjusted through the system's governance mechanisms. Hence,
+actual costs may vary depending on the current settings of these parameters.
+
+#### Estimating Gas Programmatically
+
+Users can estimate an efficient gas limit by using this function:
+
+```go
+import (
+    blobtypes "github.com/celestiaorg/celestia-app/x/blob/types"
+)
+gasLimit := blobtypes.DefaultEstimateGas([]uint32{uint32(sizeOfDataInBytes)})
+```
+
+If using a celestia-node light client, then this function is automatically
+called for you when submitting a blob. This function works by breaking down the
+components of calculating gas for a blob transaction. These components consist
+of a flat costs for all PFBs, the size of each blob and how many shares each
+uses and the parameter for gas used per byte. More information about how gas is
+used can be found in the [gas
+specs](https://github.com/celestiaorg/celestia-app/blob/d17e231ae3a0150b50a1854f3e9a268c34502b6b/specs/src/specs/resource_pricing.md)
```

Comment:
should we lock this and the next link to the commit or to a branch?

Reply:
I just did a commit for now as its not technically in a release, but we could update it in the future to a release